File: 150. Evaluate Reverse Polish Notation/main.py
from typing import List


class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = list()

        for token in tokens:
            if len(token) > 1 or token.isdigit():
                stack.append(int(token))
            else:
                num2 = stack.pop()
                num1 = stack.pop()

                if token == "+":
                    stack.append(num1 + num2)
                elif token == "-":
                    stack.append(num1 - num2)
                elif token == "*":
                    stack.append(num1 * num2)
                elif token == "/":
                    stack.append(int(num1 / num2))

        return stack.pop()


class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = list()

        for token in tokens:
            if token == "+":
                num2 = stack.pop()
                num1 = stack.pop()
                stack.append(num1 + num2)
            elif token == "-":
                num2 = stack.pop()
                num1 = stack.pop()
                stack.append(num1 - num2)
            elif token == "*":
                num2 = stack.pop()
                num1 = stack.pop()
                stack.append(num1 * num2)
            elif token == "/":
                num2 = stack.pop()
                num1 = stack.pop()
                stack.append(int(num1 / num2))
            else:
                stack.append(int(token))

        return stack.pop()


class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = list()

        for token in tokens:
            if token == "+":
                stack[-1] = stack[-2] + stack.pop()
            elif token == "-":
                stack[-1] = stack[-2] - stack.pop()
            elif token == "*":
                stack[-1] = stack[-2] * stack.pop()
            elif token == "/":
                stack[-1] = int(stack[-2] / stack.pop())
            else:
                stack.append(int(token))

        return stack.pop()

File: 150. Evaluate Reverse Polish Notation/main.cpp
#include <stack>
#include <string>
#include <vector>

using namespace std;

class Solution {
public:
    int evalRPN(vector<string> &tokens) {
        vector<int> stacks;

        for (const auto &token : tokens) {
            if (token == "+") {
                int num2 = stacks.back();
                stacks.pop_back();
                int num1 = stacks.back();
                stacks.pop_back();
                stacks.push_back(num1 + num2);
            } else if (token == "-") {
                int num2 = stacks.back();
                stacks.pop_back();
                int num1 = stacks.back();
                stacks.pop_back();
                stacks.push_back(num1 - num2);
            } else if (token == "*") {
                int num2 = stacks.back();
                stacks.pop_back();
                int num1 = stacks.back();
                stacks.pop_back();
                stacks.push_back(num1 * num2);
            } else if (token == "/") {
                int num2 = stacks.back();
                stacks.pop_back();
                int num1 = stacks.back();
                stacks.pop_back();
                stacks.push_back(num1 / num2);
            } else {
                stacks.push_back(stoi(token));
            }
        }

        return stacks.back();
    }
};

class Solution {
public:
    int evalRPN(vector<string> &tokens) {
        vector<int> stacks;

        for (const auto &token : tokens) {
            if (token.size() > 1 || isdigit(token[0])) {
                stacks.push_back(stoi(token));
            } else {
                int num2 = stacks.back();
                stacks.pop_back();
                int num1 = stacks.back();
                stacks.pop_back();

                if (token == "+") {
                    stacks.push_back(num1 + num2);
                } else if (token == "-") {
                    stacks.push_back(num1 - num2);
                } else if (token == "*") {
                    stacks.push_back(num1 * num2);
                } else if (token == "/") {
                    stacks.push_back(num1 / num2);
                }
            }
        }

        return stacks.back();
    }
};

class Solution {
public:
    int evalRPN(vector<string> &tokens) {
        stack<int> stacks;

        for (const auto &token : tokens) {
            if (token.size() > 1 || isdigit(token[0])) {
                stacks.push(stoi(token));
            } else {
                int num2 = stacks.top();
                stacks.pop();
                int num1 = stacks.top();
                stacks.pop();

                if (token == "+") {
                    stacks.push(num1 + num2);
                } else if (token == "-") {
                    stacks.push(num1 - num2);
                } else if (token == "*") {
                    stacks.push(num1 * num2);
                } else if (token == "/") {
                    stacks.push(num1 / num2);
                }
            }
        }

        return stacks.top();
    }
};